python/gui/forwarding/consumer.py

```diff
@@ -21,15 +22,38 @@
 LOGGER = logging.ConfiguredLogger()
 
 
+ASGIView = typing.Callable[
```

Comment:
Why not use a protocol instead?
```python
class ASGIView(typing.Protocol):
    def __call__(
        self,
        scope: typing.Mapping,
        receive: typing.Callable[[bool | None, float | None], typing.Mapping],
        send: typing.Callable[[typing.Mapping], None],
    ) -> typing.Coroutine[typing.Any, typing.Any, None]: ...
```

Reply:
What's the benefit of using a protocol in this instance? The purpose is to describe a hint for an expected signature for type hinting, not a promise for an object structure.

Would changing its name to `ASGIViewSignature` make more sense?